src/widget/wsearchrelatedtracksmenu.cpp

```diff
@@ -0,0 +1,291 @@
+        if (!primaryArtist.isEmpty()) {
+            // Search tracks with similar artist(s)
+            {
+                const auto actionText =
+                        tr("Artist: \"%1\"").arg(primaryArtist);
+                const QString searchQuery =
+                        QStringLiteral("artist:\"") +
+                        primaryArtist +
+                        QChar('"');
+                addSeparatorBeforeNextAction = addTriggerSearchAction(
+                        addSeparatorBeforeNextAction,
+                        actionText,
+                        searchQuery);
+            }
+            if (!secondaryArtist.isEmpty()) {
+                const auto actionText =
+                        tr("Artist: \"%1\"").arg(secondaryArtist);
+                const QString searchQuery =
+                        QStringLiteral("artist:\"") +
+                        secondaryArtist +
+                        QChar('"');
+                addSeparatorBeforeNextAction = addTriggerSearchAction(
+                        addSeparatorBeforeNextAction,
+                        actionText,
+                        searchQuery);
+            }
+            {
+                const auto actionText =
+                        tr("Album Artist: \"%1\"").arg(primaryArtist);
+                const QString searchQuery =
+                        QStringLiteral("album_artist:\"") +
+                        primaryArtist +
+                        QChar('"');
+                addSeparatorBeforeNextAction = addTriggerSearchAction(
+                        addSeparatorBeforeNextAction,
+                        actionText,
+                        searchQuery);
+            }
+            if (!secondaryArtist.isEmpty()) {
+                const auto actionText =
+                        tr("Album Artist: \"%1\"").arg(secondaryArtist);
+                const QString searchQuery =
+                        QStringLiteral("album_artist:\"") +
+                        secondaryArtist +
+                        QChar('"');
+                addSeparatorBeforeNextAction = addTriggerSearchAction(
+                        addSeparatorBeforeNextAction,
+                        actionText,
+                        searchQuery);
+            }
+        }
+    }
+    {
+        const auto composer = track.getComposer();
+        if (!composer.isEmpty()) {
+            const auto actionText =
+                    tr("Composer: \"%1\"").arg(composer);
+            const QString searchQuery =
+                    QStringLiteral("composer:\"") +
+                    composer +
+                    QChar('"');
+            addSeparatorBeforeNextAction = addTriggerSearchAction(
+                    addSeparatorBeforeNextAction,
+                    actionText,
+                    searchQuery);
+        }
+    }
+
+    // Title and grouping actions
+    addSeparatorBeforeNextAction = !isEmpty();
+    {
+        const auto title = track.getTitle();
```

Comment:
How about searching for trailing text in parentheses and adding another menu entry without that? I think that would work for the remix/original searches in most cases. For example, if the title is "Foo (Bar Remix)", add items to search both `title:"Foo (Bar Remix)"` and `title:"Foo"`

Reply:
There are too many variations. I leave this (and the following discussion) to a subsequent PR after we collected some feedback.

Reply:
okay

Reply:
Feel free to start working on a `class TitleParser` that will be needed for this purpose ;)